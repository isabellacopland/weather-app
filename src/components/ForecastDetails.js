import React from "react";
import PropTypes from "prop-types";

function ForecastDetails({ forecast }) {
  const { date, humidity, temperature, wind } = forecast;

  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__maxTemperature">
        Max: {temperature.max}&deg;C
      </div>
      <div className="forecast-details__minTemperature">
        Min: {temperature.min}&deg;C
      </div>
      <div className="forecast-details__humidity">Humidity: {humidity}%</div>
      <div className="forecast-details__wind_speed">Speed: {wind.speed}mph</div>
      <div className="forecast-details__wind_direction">
        Direction: {wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }),
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }),
    humidity: PropTypes.number,
    description: PropTypes.string,
    icon: PropTypes.string,
  }).isRequired,
};
