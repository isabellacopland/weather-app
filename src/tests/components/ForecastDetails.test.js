import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    temperature: {
      max: 22,
      min: 12,
    },
    humidity: 60,
    wind: {
      speed: 60,
      direction: "ne",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("render correct value for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText("Thu Jan 01 1970")).toHaveClass("forecast-details__date");
    expect(getByText("Max: 22°C")).toHaveClass("forecast-details__maxTemperature");
    expect(getByText("Min: 12°C")).toHaveClass("forecast-details__minTemperature");
    expect(getByText("Humidity: 60%")).toHaveClass("forecast-details__humidity");
    expect(getByText("Speed: 60mph")).toHaveClass("forecast-details__wind_speed");
    expect(getByText("Direction: ne")).toHaveClass("forecast-details__wind_direction");
  });
});
