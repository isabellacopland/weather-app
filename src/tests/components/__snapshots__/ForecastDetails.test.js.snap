// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__maxTemperature"
    >
      Max: 22°C
    </div>
    <div
      class="forecast-details__minTemperature"
    >
      Min: 12°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 60%
    </div>
    <div
      class="forecast-details__wind_speed"
    >
      Speed: 60mph
    </div>
    <div
      class="forecast-details__wind_direction"
    >
      Direction: ne
    </div>
  </div>
</DocumentFragment>
`;
